### PR TITLE
Fix find_package for benchmark

### PR DESCRIPTION
**Pull request by lukasberbuer**
**Description**: Fix find_package for benchmark as discussed here: https://github.com/microsoft/onnxruntime/pull/11146

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -707,9 +707,9 @@ endif()
 
 if (onnxruntime_BUILD_BENCHMARKS)
   if (onnxruntime_PREFER_SYSTEM_LIB)
-    find_package(benchmark)
+    find_package(benchmark CONFIG)
   endif()
-  if (benchmark_FOUND)
+  if (TARGET benchmark::benchmark)
```
Comment (JackBoosY) [inline cmake/CMakeLists.txt:712]: Should we use `benchmark::benchmark` in some `target_link_librarires`?

Comment (lukasberbuer) [reply cmake/CMakeLists.txt:712]: I only found `benchmark::benchmark` for `target_link_libraries`:
- https://github.com/microsoft/onnxruntime/blob/master/cmake/onnxruntime_unittests.cmake#L895
- https://github.com/microsoft/onnxruntime/blob/master/cmake/onnxruntime_unittests.cmake#L903

Did I miss something?

Comment (JackBoosY) [reply cmake/CMakeLists.txt:712]: Ah yes, sorry for disturb.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**